managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -480,7 +482,24 @@ public void operationComplete(ManagedCursorInfo info, Stat stat) {
                     // Need to proceed and read the last entry in the specified ledger to find out the last position
                     log.info("[{}] Consumer {} meta-data recover from ledger {}", ledger.getName(), name,
                             info.getCursorsLedgerId());
-                    recoverFromLedger(info, callback);
+                    recoverFromLedger(info, new VoidCallback() {
+                        @Override
+                        public void operationComplete() {
+                            LinkedHashSet<Long> ledgersDeletedButNotAcked = checkLedgersDeletedButNotAcked();
+                            if (!ledgersDeletedButNotAcked.isEmpty()){
+                                log.warn("[{}] ledgers {} has deleted from topic, but still in the records of"
+                                                + " incomplete-ack {}. If it is not cleaned up in time, it will cause"
+                                                + " the mark deleted position to not move forward.",
+                                        ledger.getName(), ledgersDeletedButNotAcked, name);
```

Comment:
Is it possible to add information about the topic and subscription stuck?

Reply:
We can know the topic by `ledger.getName()` and the subscription by `cursor.this.name`